pkg/k8s/watchers/watcher.go

```diff
@@ -367,9 +395,9 @@ type WatcherConfiguration interface {
 	utils.ServiceConfiguration
 }
 
-// EnableK8sWatcher watches for policy, services and endpoint changes on the Kubernetes
-// api server defined in the receiver's daemon k8sClient.
-func (k *K8sWatcher) EnableK8sWatcher(ctx context.Context) error {
+// EnableK8sWatcher watches for policy, services and endpoint changes on the
+// Kubernetes api server defined in the receiver's daemon k8sClient.
+func (k *K8sWatcher) EnableK8sWatcher(ctx context.Context, resources []string) error {
```

Comment:
Any thoughts on whether it might make sense to pass a map of handler functions instead of string slice here?

Reply:
It crossed my mind while reading this code, it might make it easier to read as opposed to long switch/case block, just thought we could discuss that :)

Reply:
I briefly thought about it, it does sound tidier. Only thing is that the function prototypes are different for the different init functions. I wasn't sure if it would make sense to force them all to conform to the same signature just for this cleanup.